packages/victory-core/src/victory-util/hooks/use-domain.ts

```diff
@@ -0,0 +1,100 @@
+import * as React from "react";
+import * as d3Array from "victory-vendor/d3-array";
+import { Axis, ForAxes, Tuple, ValueOrAxesObject } from "../../types";
+import { PaddingProps } from "../../victory-theme/victory-theme";
+import { DomainTuple, DomainValue } from "../types";
+
+interface DomainProps {
+  domain?: ValueOrAxesObject<DomainTuple>;
+  maxDomain?: ValueOrAxesObject<number>;
+  minDomain?: ValueOrAxesObject<number>;
+  domainPadding?: ValueOrAxesObject<number | Tuple<number>>;
+  width?: number;
+  height?: number;
+  padding?: PaddingProps;
```

Comment:
In a perfect world there would be one component props interface I could pick these props from, but they seem to be scattered around a few different type definitions. The structure of these types might be worth re-evaluating, even if it ends up being a breaking change.